### PR TITLE
Improve flags for better support of xlC and nvcc combinations

### PR DESCRIPTION
**Pull request by davidbeckingsale**
Ran into some issues building with xlC and nvcc - these changes fix that, by separating host and device flags, and adding support for C++ standards when xlC is used.

### CODE REVIEW

File: cmake/SetupCompilerOptions.cmake [hunk 1]
```diff
@@ -229,8 +229,18 @@ if( BLT_CXX_STD STREQUAL c++98 )
     set(CMAKE_CXX_STANDARD 98)
 elseif( BLT_CXX_STD STREQUAL c++11 )
     set(CMAKE_CXX_STANDARD 11)
+    if (COMPILER_FAMILY_IS_XL)
```
Comment (davidbeckingsale) [inline cmake/SetupCompilerOptions.cmake]: I'm all for suggestions on a more universal way to handle this. I believe that we would run into this problem with Intel also on older CMake versions.

Comment (cyrush) [inline cmake/SetupCompilerOptions.cmake]: This looks fine, it's clear you are adding flags for xl for the c++11 & c++14 case. 
If we need a fix for other cmakes thats ok as well, but our goal is to work with a few blessed versions of cmake -- currently 3.3.x and 3.8.x.

Comment (davidbeckingsale) [inline cmake/SetupCompilerOptions.cmake]: I am running into this issue with Intel also. CMAKE_CXX_STANDARD does not add the flags. I've not checked this with 3.8.x yet

Comment (white238) [inline cmake/SetupCompilerOptions.cmake]: You need to go to CMake 3.4 i think

Comment (davidbeckingsale) [inline cmake/SetupCompilerOptions.cmake]: I'm using 3.4.1 - it's not supported. We have to work around this in RAJA also.

**Pull request merged**